src/core/internal/elf/io.d

```diff
@@ -134,41 +135,70 @@ template ElfIO(Elf_Ehdr, Elf_Shdr, ubyte ELFCLASS)
             return true;
         }
 
+        /**
+         * Returns a struct to iterate over the named sections.
+         * Examples:
+         * --------------------
+         * foreach (index, name, sectionHeader; elfFile.namedSections) ...
+         * --------------------
+         */
+        NamedSections namedSections() const
+        {
+            return NamedSections(this);
+        }
+
         /**
          * Tries to find the header of the section with the specified name.
          * Returns: True on success.
          */
         bool findSectionHeaderByName(const(char)[] sectionName, out ElfSectionHeader header) const
         {
-            const index = findSectionIndexByName(sectionName);
-            if (index == -1)
-                return false;
-            header = ElfSectionHeader(this, index);
-            return true;
+            foreach (index, name, sectionHeader; namedSections)
+            {
+                if (name == sectionName)
+                {
+                    header = move(sectionHeader);
+                    return true;
+                }
+            }
+            return false;
         }
+    }
 
-        /**
-         * Tries to find the index of the section with the specified name.
-         * Returns: -1 if not found, otherwise 0-based section index.
-         */
-        size_t findSectionIndexByName(const(char)[] sectionName) const
```

Comment:
This function has been removed. These modules have just been added with v2.091 (#2330); at the time I originally worked on this (~1.5 years ago), there was no `move` in druntime (only in Phobos), so I've resorted to this little helper for `findSectionHeaderByName()`.